README.md

```diff
@@ -9,40 +9,77 @@ For further details see the [proposal](proposal.md).
-| `ImmutableMap` | `Map` | `immutableMapOf`, `immutableHashMapOf` |
-
-The default implementations of `ImmutableSet` and `ImmutableMap`, which are returned by `immutableSetOf` and `immutableMapOf`
+This library provides interfaces for immutable and persistent collections.
+
+#### Immutable collection interfaces    
+ 
+| Interface | Bases 
+| ----------| ----- 
+| `ImmutableCollection` | `Collection` |
+| `ImmutableList` | `ImmutableCollection`, `List` | 
+| `ImmutableSet` | `ImmutableCollection`, `Set` | 
+| `ImmutableMap` | `Map` |
+
+#### Persistent collection interfaces
+
+| Interface | Bases 
+| ----------| ----- 
+| `PersistentCollection` | `ImmutableCollection` | 
+| `PersistentList` | `PersistentCollection`, `ImmutableList` | 
+| `PersistentSet` | `PersistentCollection`, `ImmutableSet` | 
+| `PersistentMap` | `ImmutableMap` |
+
+#### Persistent collection builder interfaces
+
+| Interface | Bases 
+| ----------| ----- 
+| `PersistentCollection.Builder` | `MutableCollection` | 
+| `PersistentList.Builder` | `PersistentCollection.Builder`, `MutableList` | 
+| `PersistentSet.Builder` | `PersistentCollection.Builder`, `MutableSet` | 
+| `PersistentMap.Builder` | `MutableMap` |
+
+
+To instantiate an empty persistent collection or a collection with the specified elements the functions 
+`persistentListOf`, `persistentSetOf`, and `persistentMapOf` can be used.
+
+The default implementations of `PersistentSet` and `PersistentMap`, which are returned by `persistentSetOf` and `persistentMapOf`,
 preserve the element insertion order during iteration. This comes at expense of maintaining more complex data structures.
-If the order of elements doesn't matter, more efficient `immutableHashSetOf` and `immutableHashMapOf` could be used.
+If the order of elements doesn't matter, the more efficient implementations returned by the functions 
+`persistentHashSetOf` and `persistentHashMapOf` could be used.
 
 ### Operations
 
 #### toImmutableList/Set/Map
+
 Converts a read-only or mutable collection to an immutable one.
-If the receiver is already immutable and has the required type, returns itself.
+If the receiver is already immutable and has the required type, returns it as is.
 
 ```kotlin
 fun Iterable<T>.toImmutableList(): ImmutableList<T>
 fun Iterable<T>.toImmutableSet(): ImmutableSet<T>
 ```
 
+#### toPersistentList/Set/Map
+
+Converts a read-only or mutable collection to a persistent one.
+If the receiver is already persistent and has the required type, returns it as is.
+If the receiver is a builder of the required persistent collection type, calls `build` on it and returns the result.
+
+```kotlin
+fun Iterable<T>.toPersistentList(): PersistentList<T>
+fun Iterable<T>.toPersistentSet(): PersistentSet<T>
+```
+
 #### `+` and `-` operators
 
-`plus` and `minus` operators on immutable collections exploit their immutability
+`plus` and `minus` operators on persistent collections exploit their immutability
```

Comment:
Not in domain of this PR, but having `plus`/`minus` operators on immutable collections to return immutable collections would make the immutable interfaces more useful.